tests/api/rooms.test.js

```diff
@@ -98,18 +98,242 @@ describe("Room Test /api/rooms", () => {
+      expect(song.artist.length).toBeGreaterThan(1);
+    });
+  });
+});
+
+describe("POST /api/rooms/new", () => {
+  let addedSong;
+
+  beforeAll(async () => {
+    await mongoMemoryServer.connect();
+
+    const testSong = {
+      title: "Billie Jean",
+      audioURL: "https://bucketname.s3.location.amazonaws/music/billiejean",
+      imageURL: "https://bucketname.s3.location.amazonaws/image/billiejean",
+      artist: "Micheal Jackson",
+    };
+
+    await Song.create(testSong);
+
+    addedSong = await Song.findOne({
+      title: "Billie Jean",
+    });
+  });
+
+  afterAll(async () => {
+    await mongoMemoryServer.closeDatabase();
+  });
+
+  afterEach(async () => {
+    await BattleRoom.deleteMany({});
+  });
+
+  it("creates a new battle room with valid request data and returns 201 status code", async () => {
+    const roomListBefore = await BattleRoom.find({});
+
+    const newRoom = {
+      song: addedSong._id,
+      createdBy: "artist",
+      uid: "m5zw3tb2fqpfeqjmg97twmyhlgq3",
+    };
+
+    const response = await request(appTest)
+      .post("/api/rooms/new")
+      .send(newRoom);
+
+    const roomListAfter = await BattleRoom.find({});
+
+    const addedRoom = await BattleRoom.find({
+      uid: newRoom.uid,
+    });
+
+    expect(response.status).toEqual(201);
+    expect(addedRoom).toBeDefined();
+    expect(roomListAfter.length).toEqual(roomListBefore.length + 1);
+  });
+
+  it("returns a 400 status when the request body is missing a field", async () => {
+    const newRoom = {
+      song: addedSong._id,
+      createdBy: "artist",
+    };
+
+    const response = await request(appTest)
+      .post("/api/rooms/new")
+      .send(newRoom);
+
+    expect(response.status).toEqual(400);
+    expect(response.body).toEqual({ message: "Invalid requested body" });
+  });
+});
+
+describe("GET /api/rooms/:roomId", () => {
+  let idFound;
+
+  beforeAll(async () => {
+    await mongoMemoryServer.connect();
+    const sampleSong = {
+      title: "Billie Jean",
+      audioURL: "https://bucketname.s3.location.amazonaws/music/billiejean",
+      imageURL: "https://bucketname.s3.location.amazonaws/image/billiejean",
+      artist: "Micheal Jackson",
+    };
+
+    await Song.create(sampleSong);
+
+    const addedSong = await Song.findOne({
+      title: sampleSong.title,
+    });
+
+    const newNote = {
+      title: sampleSong.title,
+      note: [
+        {
+          time: 0.2,
+          key: "s",
+          positionY: 0,
+        },
+        {
+          time: 0.3,
+          key: "d",
+          positionY: 0,
+        },
+      ],
+    };
+
+    await Note.create(newNote);
+
+    const newRoom = {
+      song: addedSong._id,
+      createdBy: "artist",
+      uid: "m5zw3tb2fqpfeqjmg97twmyhlgq3",
+    };
+
+    await BattleRoom.create(newRoom);
+
+    const roomIdFound = await BattleRoom.findOne({
+      uid: "m5zw3tb2fqpfeqjmg97twmyhlgq3",
+    });
+
+    idFound = roomIdFound._id.toString();
```

Comment:
```suggestion
this.idFound = roomIdFound._id.toString();
```
`let`으로 선언된 변수 대신 `this.idFound`를 사용하면 `describe` 블록의 컨텍스트에 값을 저장하고 테스트 케이스 내에서 액세스할 수 있습니다.